src/display/text_layer.js

```diff
@@ -185,10 +185,11 @@ function appendText(task, geom, styles, ctx) {
   let shouldScaleText = false;
   if (
     geom.str.length > 1 ||
-    (task._enhanceTextSelection && AllWhitespaceRegexp.test(geom.str))
+    (task._enhanceTextSelection &&
+      (geom.isOneSpace || AllWhitespaceRegexp.test(geom.str)))
   ) {
     shouldScaleText = true;
-  } else if (geom.transform[0] !== geom.transform[3]) {
+  } else if (!geom.isOneSpace && geom.transform[0] !== geom.transform[3]) {
```

Comment:
Again, I might be missing something, but do we really need this new `geom.isOneSpace` property?
Could we simply not replace that part with a `geom.str !== " " && ...` check, and avoid having to add a new property to each textItem?